source/common/filter/config_discovery_impl.h

```diff
@@ -192,7 +192,8 @@ class HttpDynamicFilterConfigProviderImpl
     auto* factory =
         Registry::FactoryRegistry<Server::Configuration::NamedHttpFilterConfigFactory>::getFactory(
             factory_name);
-    const bool is_terminal_filter = factory->isTerminalFilterByProto(message, factory_context_);
+    const bool is_terminal_filter =
+        factory->isTerminalFilterByProto(message, factory_context_.getServerFactoryContext());
```

Comment:
Thanks, this is great for ECDS. We can erase a type template parameter later if all filter factories depend on server context rather than factory context.